build/commands/lib/config.js

```diff
@@ -398,14 +400,17 @@ Config.prototype.buildArgs = function () {
   }
 
   if (this.targetArch === 'x64' &&
-    process.platform === 'linux' &&
+    (this.targetOS || process.platform) === 'linux' &&
```

Comment:
I believe this can be simplified to just `this.targetArch === 'x64' && this.targetOS === 'linux'` since we're already excluding android

Reply:
also if we're cross compiling macos on linux, I believe this would incorrectly set use_vaapi to true

Reply:
actually this whole vaapi thing looks suspect to me and I think we need to revisit it

Reply:
also pretty sure this should be `use_vaapi_x11`, not `use_vaapi`

```
# VA-API should also be compiled by default on x11-using linux devices
  # using x86/x64.
  use_vaapi = use_vaapi_x11 || (is_chromeos_lacros &&
                                (target_cpu == "x86" || target_cpu == "x64"))
```

Reply:
I fixed the `if` statement as you suggested in f2a3c0b1ac727477d22ac1984469af4c26111089.

> if we're cross compiling macos on linux, I believe this would incorrectly set use_vaapi to true

I actually don't think the implementation you reviewed would have done this. The reason is that `targetOS` is set to `mac` when we cross-compile.

> also pretty sure this should be use_vaapi_x11, not use_vaapi

It's possible. I don't yet understand why we need to set `use_vaapi` or `use_vaapi_x11` at all, given that they have dynamic defaults that appear reasonable.

```
  # Indicates if X11 VA-API-based hardware acceleration is to be used.
  # See also the comment near the |use_vaapi| arg.
  use_vaapi_x11 =
      is_linux && ozone_platform_x11 && !is_chromecast && !is_chromeos_lacros &&
      (target_cpu == "x86" || target_cpu == "x64")
```

The approach I'd choose is to create a follow-up issue to investigate this.